kotlin/query-with-cql-filters/src/main/java/com/esri/arcgisruntime/sample/querywithcqlfilters/MainActivity.kt

```diff
@@ -0,0 +1,374 @@
+                    "Failed to load OGC Feature Collection Table" +
+                            ogcFeatureCollectionTable.loadError.message,
+                    Toast.LENGTH_LONG
+                ).show()
+                Log.e(
+                    "OGC Load error: ",
+                    ogcFeatureCollectionTable.loadError.message.toString()
+                )
+            }
+        }
+
+        // Load the table
+        ogcFeatureCollectionTable.loadAsync()
+
+        // Handles CQL Filters in a Bottom Sheet View.
+        fab.setOnClickListener {
+            openBottomSheetView()
+        }
+    }
+
+    /**
+     * Handles CQL Filters in a Bottom Sheet View.
+     */
+    private fun openBottomSheetView() {
+
+        // Resets views in bottomSheet to default values.
+        resetBottomSheetValues()
+
+        // Creates a new BottomSheetDialog
+        val dialog = BottomSheetDialog(this)
+
+        // Inflates layout file
+        val cqlFiltersLayoutBinding = CqlFiltersLayoutBinding.inflate(layoutInflater)
+
+        // Sets the Where Clause for CQL filter
+        cqlFiltersLayoutBinding.whereClauseLayout.setOnClickListener {
+
+            // Creates a dialog to choose a where clause
+            val alertDialog: AlertDialog.Builder = AlertDialog.Builder(this)
+            alertDialog.setTitle("AlertDialog")
+            val checkedItem = cqlQueryListPosition
+
+            alertDialog.setSingleChoiceItems(
+                cqlQueryList.toTypedArray(), checkedItem
+            ) { dialog, which ->
+
+                // Updates the selected where clause
+                cqlQueryListPosition = which
+                cqlFiltersLayoutBinding.cqlQueryTextView.text = cqlQueryList[which]
+
+                // Dismiss dialog
+                dialog.dismiss()
+            }
+
+            // Displays the where clause dialog
+            val alert: AlertDialog = alertDialog.create()
+            alert.show()
+        }
+
+        // Sets the view to the default value of max features (1000)
+        cqlFiltersLayoutBinding.maxFeaturesEditText.setText(maxFeatures.toString())
+
+        // Sets from date to Jun-13-2011 by default
+        cqlFiltersLayoutBinding.fromDatePicker.updateDate(2011, 5, 13)
+        // Sets to date to Jan-7-2012 by default
+        cqlFiltersLayoutBinding.toDatePicker.updateDate(2012, 0, 7)
+
+        // Sets up filters for the query when Apply is clicked.
+        cqlFiltersLayoutBinding.applyTv.setOnClickListener {
+
+            // Retrieves the max features
+            val maxFeaturesText =
+                cqlFiltersLayoutBinding.maxFeaturesEditText.text.toString()
+            maxFeatures = when {
+                maxFeaturesText == "" -> 1000
+                maxFeaturesText.toInt() <= 0 -> 1000
+                else -> maxFeaturesText.toInt()
+            }
+
+            // Retrieves if date filter is selected
+            isDateFilter = cqlFiltersLayoutBinding.dateSwitch.isChecked
```

Comment:
perhaps you could just instantiate the boolean here rather than at the top of the code: keep the reference to it at the start but instantiate it here?